### ISSUE TITLE
K8s sometimes throws error "invalid pod status" when submitting many pods

### ISSUE THREAD

**Issue by bentsherman** [nextflow-io/nextflow#1223]
### Expected behavior and actual behavior

When submitting many jobs using k8s executor, some pods may not be scheduled immediately due to insufficient resources. In this case, nextflow should be able to wait while the pod is scheduled. However, recently I've noticed that if I increase the parallelism such that many pods are pending, nextflow throws an error which I traced to this line of code:

https://github.com/nextflow-io/nextflow/blob/b49e6d74dae782def6be2b087d06fdd8fa6b9291/modules/nextflow/src/main/groovy/nextflow/k8s/client/K8sClient.groovy#L260

Here is the status object that nextflow tried to parse:
```
      "status": {
          "phase": "Pending",
          "qosClass": "Guaranteed"
      }
```

Looking at the code I can see that nextflow was expecting to see a `conditions` array which is missing. I suppose the assumption was that this conditions array would be initialized when the phase was set to `Pending`, however my experience suggests otherwise.

Although I'm not an expert on k8s internals, I think we should simply modify this `podState` function to not fail if the `conditions` array is missing:
```groovy
        if( status?.phase == 'Pending' ) {
            if( status.conditions instanceof List ) {
                final allConditions = status.conditions as List<Map>
                final cond = allConditions.find { cond -> cond.type == 'PodScheduled' }
                if( cond.reason == 'Unschedulable' ) {
                    def message = "K8s pod cannot be scheduled"
                    if( cond.message ) message += " -- $cond.message"
                    //def cause = new K8sResponseException(resp)
                    log.warn1(message)
                }
            }
            // undetermined status -- return an empty response
            return Collections.emptyMap()
        }

        throw new K8sResponseException("K8s invalid pod status (missing container status)", resp)
```

### Steps to reproduce the problem

Run a workflow which submits many more pods than the capacity of the k8s cluster, such that several pods are pending.

### Environment 

* Nextflow version: 19.04.0
* Java version: 1.8
* Operating system: Linux

**Comment by olifly** [comment]
This bug had been cropping up fairly regularly for us and I had made a quick fix, but hadn't submitted it upstream til now: https://github.com/nextflow-io/nextflow/pull/1242

**Comment by bentsherman** [comment]
Fixed in 19.07.0.
